src/components/ImageSettingsModal.jsx

```diff
@@ -15,38 +15,63 @@ export default class ImageSettingsModal extends Component {
       newFileName: fileName,
       sha: '',
       content: null,
+      isUpload: null,
     };
   }
 
   async componentDidMount() {
-    const { match, image: { fileName } } = this.props;
+    const { match, image, isUpload } = this.props;
     const { siteName } = match.params;
-    const { data: { sha, content } } = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/images/${fileName}`, {
-      withCredentials: true,
-    });
-    this.setState({ sha, content });
+    if (isUpload) {
+      const { content } = image;
+      this.setState({ content, isUpload });
+    } else {
+      const { fileName } = image;
+      const { data: { sha, content } } = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/images/${fileName}`, {
+        withCredentials: true,
+      });
+      this.setState({ sha, content, isUpload });
```

Comment:
Since you're not modifying the value of `isUpload` when it is passed down as a prop, it's redundant to to set it in state when you can access the same value from props

Reply:
Fixed in 3ea87ca